datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/SchemaTab.tsx

```diff
@@ -44,13 +44,23 @@ const LoadingWrapper = styled.div`
 export const SchemaTab = ({ properties }: { properties?: any }) => {
     const entityRegistry = useEntityRegistry();
     const baseEntity = useBaseEntity<GetDatasetQuery>();
+    const [showRaw, setShowRaw] = useState(false);
     // Dynamically load the schema + editable schema information.
-    const { entityWithSchema, loading, refetch } = useGetEntityWithSchema();
+    const { entityWithSchema, loading, refetch } = useGetEntityWithSchema(undefined, (data) => {
+        const schemaMetadata = data?.dataset?.schemaMetadata;
+        if (
+            schemaMetadata?.fields?.length === 0 &&
```

Comment:
I used similar logic to what is used here: https://github.com/datahub-project/datahub/pull/11515/files#diff-188d894d1a9f5a0f3be93c869ec3fc09abc284edb44483e6dc738e295b353273L59-L64.
Not sure if we can/should use the memo function in here. Also, do you know why there's a check for `__typename === 'TableShema'` here? This is not fully clear to me. We're storing a KafkaSchema with `documentSchema` set as the raw schema. But somehow this works. Can someone explain?